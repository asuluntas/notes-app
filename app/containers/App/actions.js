import {
  LOAD_NOTES,
  LOAD_NOTES_SUCCESS,
  LOAD_NOTES_ERROR,
} from '../Notes/constants';

import {
  ADD_NOTE,
  ADD_NOTE_SUCCESS,
  ADD_NOTE_ERROR,
} from '../HomePage/constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes, note) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
    note,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}

export function addNote(note) {
  return {
    type: ADD_NOTE,
    note,
  };
}

export function noteAdded(note) {
  return {
    type: ADD_NOTE_SUCCESS,
    note,
  };
}

export function noteAddingError(error) {
  return {
    type: ADD_NOTE_ERROR,
    error,
  };
}
