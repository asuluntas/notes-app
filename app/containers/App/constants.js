export const LOAD_NOTES = 'boilerplate/App/LOAD_NOTES';
export const LOAD_NOTES_SUCCESS = 'boilerplate/App/LOAD_NOTES_SUCCESS';
export const LOAD_NOTES_ERROR = 'boilerplate/App/LOAD_NOTES_ERROR';
export const ADD_NOTE = 'boilerplate/App/ADD_NOTE';
export const ADD_NOTE_SUCCESS = 'boilerplate/App/ADD_NOTE_SUCCESS';
export const ADD_NOTE_ERROR = 'boilerplate/App/ADD_NOTE_ERROR';
