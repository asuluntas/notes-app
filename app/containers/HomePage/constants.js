export const CHANGE_NOTE = 'boilerplate/Home/CHANGE_NOTE';
export const ADD_NOTE = 'boilerplate/Home/ADD_NOTE';
export const ADD_NOTE_SUCCESS = 'boilerplate/Home/ADD_NOTE_SUCCESS';
export const ADD_NOTE_ERROR = 'boilerplate/Home/ADD_NOTE_ERROR';
