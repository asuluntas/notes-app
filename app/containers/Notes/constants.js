export const LOAD_NOTES = 'boilerplate/Notes/LOAD_NOTES';
export const LOAD_NOTES_SUCCESS = 'boilerplate/Notes/LOAD_NOTES_SUCCESS';
export const LOAD_NOTES_ERROR = 'boilerplate/Notes/LOAD_NOTES_ERROR';
