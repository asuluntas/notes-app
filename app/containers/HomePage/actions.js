import { CHANGE_NOTE } from './constants';

export function changeNote(note) {
  return {
    type: CHANGE_NOTE,
    note,
  };
}
