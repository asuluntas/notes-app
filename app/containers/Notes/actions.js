import { LOAD_NOTES, LOAD_NOTES_SUCCESS, LOAD_NOTES_ERROR } from './constants';

export function loadNotes() {
  return {
    type: LOAD_NOTES,
  };
}

export function notesLoaded(notes) {
  return {
    type: LOAD_NOTES_SUCCESS,
    notes,
  };
}

export function notesLoadingError(error) {
  return {
    type: LOAD_NOTES_ERROR,
    error,
  };
}
