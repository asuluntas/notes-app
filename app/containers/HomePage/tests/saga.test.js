import { put, takeEvery } from 'redux-saga/effects';

import { ADD_NOTE } from '../constants';
import { noteAdded, noteAddingError } from '../actions';

import watchAddNotes, { addNote } from '../saga';

const note = 'a test note';

/* eslint-disable redux-saga/yield-effects */
describe('addNote Saga', () => {
  let addNoteGenerator;

  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    addNoteGenerator = addNote();

    const selectDescriptor = addNoteGenerator.next().value;
    expect(selectDescriptor).toMatchSnapshot();

    const callDescriptor = addNoteGenerator.next(note).value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the noteAdded action if it requests the data successfully', () => {
    const response = {
      id: 3,
      text: 'a test note',
    };
    const putDescriptor = addNoteGenerator.next(response).value;
    expect(putDescriptor).toEqual(put(noteAdded(response.text)));
  });

  it('should call the noteAddingError action if the response errors', () => {
    const response = new Error('Some error');
    const putDescriptor = addNoteGenerator.throw(response).value;
    expect(putDescriptor).toEqual(put(noteAddingError(response)));
  });
});

describe('watchAddNotesSaga Saga', () => {
  const watchAddNotesSaga = watchAddNotes();

  it('should start task to watch for ADD_NOTE action', () => {
    const takeLatestDescriptor = watchAddNotesSaga.next().value;
    expect(takeLatestDescriptor).toEqual(takeEvery(ADD_NOTE, addNote));
  });
});
