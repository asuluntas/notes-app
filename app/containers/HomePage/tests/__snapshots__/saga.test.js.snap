// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addNote Saga should call the noteAddingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`addNote Saga should call the noteAddingError action if the response errors 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/addNote",
      Object {
        "body": "{\\"text\\":\\"a test note\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`addNote Saga should dispatch the noteAdded action if it requests the data successfully 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [],
    "selector": [Function],
  },
  "type": "SELECT",
}
`;

exports[`addNote Saga should dispatch the noteAdded action if it requests the data successfully 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "/addNote",
      Object {
        "body": "{\\"text\\":\\"a test note\\"}",
        "headers": Object {
          "Accept": "application/json",
          "Content-Type": "application/json",
        },
        "method": "POST",
      },
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;
